mlflow/server/handlers.py

```diff
@@ -1131,15 +1130,15 @@ def search_datasets_handler():
 
 @catch_mlflow_exception
 def gateway_proxy_handler():
-    args = request.json
-
     gateway_uri = MLFLOW_GATEWAY_URI.get()
-    _logger.info("gateway_uri: %s", gateway_uri)
     if not gateway_uri:
-        raise MlflowException(
-            message="MLFLOW_GATEWAY_URI environment variable must be set.",
-            error_code=INTERNAL_ERROR,
-        )
+        # Pretend an empty gateway service is running
+        return {"routes": []}
```

Comment:
@prithvikannan Added this change to suppress the following errors (caused by `gateway-proxy` requests): 

<img width="1370" alt="Screen Shot 2023-09-05 at 14 02 35" src="https://github.com/mlflow/mlflow/assets/17039389/422b9350-91e7-4053-ad41-9846e7e88fd1">

Reply:
Creating fake gateway routes might be easier than the current approach.

Reply:
@harupy @prithvikannan I'm not sure that this worked. I opened the UI from the current PR branch without the Gateway env var set and saw this issue:

<img width="981" alt="Screenshot 2023-09-05 at 12 14 13 PM" src="https://github.com/mlflow/mlflow/assets/39497902/48b9af33-5161-43be-bf8d-3fb712fb9947">

I also think we should make sure the UI loads properly even if Gateway responses aren't valid / can't be obtained. Having the whole OSS MLflow UI take a dependency on Gateway correctness is pretty scary since Gateway is an experimental feature. If Gateway is unavailable, I would expect that only the prompt engineering UI components don't load / work.

Can we address this?

Reply:
Got it, should we surface the error on UI or swallow?

Reply:
The easiest solution here is just to `dispatch(searchModelGatewayRoutesApi()).catch(console.error)`.

Reply:
Surfacing seems to make sense to me :)

Reply:
Pushed https://github.com/mlflow/mlflow/pull/9503/commits/97c80bdebf36a233ce2f3d3fde909377772b2aa5

Reply:
https://github.com/mlflow/mlflow/assets/17039389/5dc4231d-7fc1-49a7-9193-16e2d05cd0b6

Reply:
and no notifications when `MLFLOW_GATEWAY_URI`:

<img width="1850" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/602cbee6-e49a-4604-ab52-88aeed888cf7">

Reply:
thanks @harupy